### PR TITLE
Romerol Zombies: Hands Edition (it's worse than borg hands! I can't believe she's done this!)

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Moves zombie infection to a species unarmed attack proc. For zombies to infect you, they must grab you and then attack you with an unarmed attack. They also bite you if you are unable to protect yourself. If you have 50 or more bio armor, congrats! You're safe from those dastardly zombie bites!

However, they will now almost certainly beat you to death. Because they hit quite hard.

Zombies also possess TRAIT_ACCURATE_PUNCHER. This means they only have a flat 10% miss chance on their hits. This encourages zombies to use their hands. They're pretty brutal after all. (Also adds a useful trait for making people better at punchies)

10-20 punches, with a stun threshold of 16. They hurt. A lot. But their reliance on punches means you can use a stool to protect yourself from zombie attacks. The mental image was funny to me.

Zombies no longer have innate armor. That was really weird and probably some outdated shit. _A lot of species code is, actually, but the result was that not only did zombies regenerate very quickly, but just took significantly less damage from everything._

Romerol zombies have usable hands, but they're stupid, incapable of complex tool use and clumsy, as well as completely incapable of using guns. I am fairly certain they cannot use things like consoles and the like. (there is also a new interaction with vendors if you're as stupid as a zombie)

This means zombies are more likely to get stranded in various parts of the station without equipment, but they can bash objects against windows and the like. This also means zombies can do wonderful things like use some objects they couldn't before. Hopefully their variety of traits prevents them from most _complicated_ mischief, but it is a brave new world for zombie kind. Full of object based opportunity. (Clumsy prevents the vast majority of truly outrageous combinations, like deswords, from being useful)

Zombies don't have a glove slot. **Zombies are deprived of their golden glubs.** _And krav maga and tackle gloves and gloves of the northstar._

_Although zombies can't bite with a martial art anyway so it's largely pointless but yes, a martial artist would retain their martial arts as a zombie._

_If it ever comes up, I'm sorry._

Zombies have hair?

## Why It's Good For The Game

So people were telling me that because zombie hands, the weapon, has some really janky shit going on with it, you could basically always infect people because infection was tied into the after_attack and didn't check for hand use, if the attack hit or not OR if the attack happened at all. That'd be an easy fix that isn't as complicated or controversial, buuuuut.....

It kind of struck me as really weird they were unarmed but not unarmed, and while the limitation prevented them from using an assortment of equipment, and enabled them to smash down airlocks and the like, practically allowing a zombie absolute freedom to go wherever, maybe this would be slightly more interesting. (Also not entirely a departure from Romero's depiction of the undead as I'm lead to believe, so maybe this is closer to the namesake)
## Changelog
:cl:
add: Romerol zombies are very, very different! They now have obtained some limited object use, and a bite! And hair!
balance: Zombies are, however, still incredibly stupid. But they have hands!
balance: Zombies are easily thwarted by vending machines. And hacking airlocks. And surgery.
balance: Zombies bash down windows and airlocks. And vending machines, but don't do that. It'll end badly.
balance: Dumb people (like zombies) are very much impressed with mechs, but can't seem to work out they're supposed to climb inside.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
it's worse than borg hands! I can't believe she's done this!

**Comment by EdgeLordExe** [comment]
> it's worse than borg hands! I can't believe she's done this!
it's worse than borg hands! I can't believe she's done this!

**Comment by TiviPlus** [comment]
> > it's worse than borg hands! I can't believe she's done this!
> 
> it's worse than borg hands! I can't believe she's done this!

it's worse than borg hands! I can't believe she's done this!

**Comment by MrDoomBringer** [comment]
> > > it's worse than borg hands! I can't believe she's done this!
> > 
> > 
> > it's worse than borg hands! I can't believe she's done this!
> 
> it's worse than borg hands! I can't believe she's done this!

it's worse than borg hands! I can't believe she's done this!

**Comment by TemporalOroboros** [comment]
> it's worse than borg hands! I can't believe she's done this!

it's worse than borg hands! I can't believe she's done this!

**Comment by TemporalOroboros** [comment]
>>>> it's worse than borg hands! I can't believe she's done this!
>>>
>>> it's worse than borg hands! I can't believe she's done this!
>>
>> it's worse than borg hands! I can't believe she's done this!
>
> it's worse than borg hands! I can't believe she's done this!

it's worse than borg hands! I can't believe she's done this!

**Comment by ATH1909** [comment]
> > > > > it's worse than borg hands! I can't believe she's done this!
> > > > 
> > > > 
> > > > it's worse than borg hands! I can't believe she's done this!
> > > 
> > > 
> > > it's worse than borg hands! I can't believe she's done this!
> > 
> > 
> > it's worse than borg hands! I can't believe she's done this!
> 
> it's worse than borg hands! I can't believe she's done this!

it's worse than borg hands! I can't believe she's done this!

**Comment by ShibaInuLord** [comment]
> 
> 
> > > > > > it's worse than borg hands! I can't believe she's done this!
> > > > > 
> > > > > 
> > > > > it's worse than borg hands! I can't believe she's done this!
> > > > 
> > > > 
> > > > it's worse than borg hands! I can't believe she's done this!
> > > 
> > > 
> > > it's worse than borg hands! I can't believe she's done this!
> > 
> > 
> > it's worse than borg hands! I can't believe she's done this!
> 
> it's worse than borg hands! I can't believe she's done this!
it's worse than borg hands! I can't believe she's done this!

**Comment by GuillaumePrata** [comment]
> > > > > > > it's worse than borg hands! I can't believe she's done this!
> > > > > > 
> > > > > > 
> > > > > > it's worse than borg hands! I can't believe she's done this!
> > > > > 
> > > > > 
> > > > > it's worse than borg hands! I can't believe she's done this!
> > > > 
> > > > 
> > > > it's worse than borg hands! I can't believe she's done this!
> > > 
> > > 
> > > it's worse than borg hands! I can't believe she's done this!
> > 
> > 
> > it's worse than borg hands! I can't believe she's done this!
> 
> it's worse than borg hands! I can't believe she's done this!

it's worse than borg hands! I can't believe she's done this!

**Comment by Couls** [comment]
> > > > > > > > it's worse than borg hands! I can't believe she's done this!
> > > > > > > 
> > > > > > > 
> > > > > > > it's worse than borg hands! I can't believe she's done this!
> > > > > > 
> > > > > > 
> > > > > > it's worse than borg hands! I can't believe she's done this!
> > > > > 
> > > > > 
> > > > > it's worse than borg hands! I can't believe she's done this!
> > > > 
> > > > 
> > > > it's worse than borg hands! I can't believe she's done this!
> > > 
> > > 
> > > it's worse than borg hands! I can't believe she's done this!
> > 
> > 
> > it's worse than borg hands! I can't believe she's done this!
> 
> it's worse than borg hands! I can't believe she's done this!

it's worse than borg hands! I can't believe she's done this!

**Comment by Krysonism** [comment]
How will zombies prevent getting bolted in or otherwise gbj'd?

**Comment by necromanceranne** [comment]
Rot I guess.

**Comment by MrDoomBringer** [comment]
Doesn't sound like particularly fun gameplay on the zombie's end - should probably let them very very slowly pry open or bash down airlocks - via snowflake code if necessary ig

**Comment by necromanceranne** [comment]
The fact someone can lock you down in a spot with no way out is the fault of much wider design. I get that this is especially bad for zombies given their lack of advanced tool use but, not everyone is waltzing around with a full suite of tools and those guys are just as screwed. If the AI spots you and you have no tools, and then door bolts and depowers the room you're in, and you lack a blunt object to smash at least a window with assuming you have a window to escape through, looking at various parts of meta and delta for this), that's just how that goes. We can't expect everyone to be capable of getting out of that situation, so why should a zombie, a braindead predator, be able to accomplish what a greyshirt cannot?

It might not be fun for the zombies, but it's just as not fun for anyone. The inclusion of hands was at most hopefully going to make up for the loss of what is essentially a free pass to anywhere short of a location being blocked by a wall.

Maybe it's time to look at adding safety doors to more areas of the station. Like the public ones. Also, opening firelocks, though I believe the weedman himself vetoed making firelocks less unreasonable to escape through.

**Comment by necromanceranne** [comment]
Actually I just had a look. I am incorrect I actually believe they are capable of basic tool use as of right now. The trait and IsAdvancedToolUser doesnt even seem to cover hacking. So if they do have tools, they can use them. 

That's kind of weird to be honest. I can get them being able to use crowbars but multitooling an airlock or APC? Eeeeh...

Tampering with that will have much wider consequences and I would much prefer not snowflaking this. So I might atomize it.

**Comment by itseasytosee** [comment]
Practically I see this as a straight nerf. Zombies get very little benefit from this.

**Comment by tralezab** [comment]
Zombies no longer breaking down airlocks but instead sheepishly screwdrivering it open and hacking it? /really/?

**Comment by IndieanaJones** [comment]
This is pretty lame from a gameplay perspective.

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -889,6 +889,9 @@
 	if(HAS_TRAIT(user, TRAIT_PRIMITIVE)) //no lavalizards either.
 		to_chat(user, "<span class='warning'>The knowledge to use this device eludes you!</span>")
 		return
+	if(HAS_TRAIT(user, TRAIT_DUMB)) //braindead idiots don't get in the mech
```
Comment (Tlaltecuhtli) [inline code/game/mecha/mecha.dm:892]: this line of code is offensive

**Comment by Tlaltecuhtli** [comment]
so zombies hack and punch people ? might as well make zombies wear a grey jumpsuit and a gasmask

**Comment by Doosteroni** [comment]
So it turns people into greytiders? Bummer

File: code/_globalvars/traits.dm [hunk 2]
```diff
@@ -83,6 +83,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_ABDUCTOR_SCIENTIST_TRAINING" = TRAIT_ABDUCTOR_SCIENTIST_TRAINING,
 		"TRAIT_SURGEON" = TRAIT_SURGEON,
 		"TRAIT_STRONG_GRABBER" = TRAIT_STRONG_GRABBER,
+		"TRAIT_STRONG_PUNCHER" = TRAIT_STRONG_PUNCHER,
```
Comment (Fikou) [inline code/_globalvars/traits.dm:86]: shouldnt this be accurate puncher or something

**Comment by Winterous** [comment]
Ok I REALLY like the ideas of...
1. Making infection not automatic if you hit someone at all, the grab+hit method is good and makes sense.
2. Making chairs block their attacks, because unarmed.

File: code/game/mecha/mecha.dm (hunk 1 above)
Comment (Krysonism) [reply code/game/mecha/mecha.dm:892]: This snippet of code is a port from girlystation

https://github.com/tgstation/tgstation/pull/45017

**Comment by Tharcoonvagh** [comment]
Dont let them change their clothes, I dont wanna see armored zombies running around

**Comment by Tharcoonvagh** [comment]
Overall there are way too many balance issues, if they can hack airlocks you lost me

File: code/_globalvars/traits.dm (hunk 2 above)
Comment (necromanceranne) [reply code/_globalvars/traits.dm:86]: Yeah ryll brought that up actually. I'll change it to a more accurate trait name (haha get it)

**Comment by Couls** [comment]
While it'd be interesting seeing zombies shamble around with crowbars in their hands or other miscellaneous objects, it's been proven time and time again that players can't be trusted with this power and as-is this seems like it would just be more annoying but I do like the idea of zombies grabbing people to infect, it would be interesting if resisting out of the grab took stamina as well so multiple grabs would deplete stamina rapidly but ensure a single grab isn't an instant win

**Comment by necromanceranne** [comment]
That's actually how grabs already work, but grabbing is very all or nothing in what happens when you are grabbed. You lose hand use and drop your stuff so it can be a pretty binary "You lose" scenario.

Also, I'm gonna add some more restrictions to zombos. I'm thinking adding some ability to smash windows so they can at least beat down a window with enough time and effort, but also make them unable to use the more complex tools available to them. Screwdrivers take a few attempts, and they can't use multitools or hack whatsoever. Also maybe they drop stuff a lot. 

So they can get through parts of the station via breaking stuff but not through hacking.

**Comment by IndieanaJones** [comment]
> 
> 
> That's actually how grabs already work, but grabbing is very all or nothing in what happens when you are grabbed. You lose hand use and drop your stuff so it can be a pretty binary "You lose" scenario.
> 
> Also, I'm gonna add some more restrictions to zombos. I'm thinking adding some ability to smash windows so they can at least beat down a window with enough time and effort, but also make them unable to use the more complex tools available to them. Screwdrivers take a few attempts, and they can't use multitools or hack whatsoever. Also maybe they drop stuff a lot.
> 
> So they can get through parts of the station via breaking stuff but not through hacking.
So what's the point of these changes?  They aren't really making zombie outbreaks any more fun by making the zombies into bootleg assistants, and with the nerfs, they should be able to be put down rather easily.  Seems like a losing situation all the way around.

**Comment by necromanceranne** [comment]
It just was weird that zombies hit things with a weapon that was forced into their hands, and I remember several instances of that breaking and resulting in a lot of weird stuff, like summoned items and other things being forced into their hands while they have their zombie claws and usually deleting the claws so they end up with a free hand anyway. And in general, the fact that they claw you to infect just not really sitting right.

I felt if most of the zombie stuff was attached to them as a species over a weapon, maybe zombies will be at least....a little less janky.

I acknowledge this is a netloss in just sheer braindead lethality, and for the most part that's what people care about, but if I can add a few more interactions that will make them more zombie-like, and fix the problems with them potentially being stranded, hopefully it will be a relatively neutral change.

**Comment by Tharcoonvagh** [comment]
Keep in mind that zombies are currently bugged so if they die (in their zombie form) they revert to a human corpse and dont revive

**Comment by Kaffe-work** [comment]
> The fact someone can lock you down in a spot with no way out is the fault of much wider design. I get that this is especially bad for zombies given their lack of advanced tool use but, not everyone is waltzing around with a full suite of tools and those guys are just as screwed. If the AI spots you and you have no tools, and then door bolts and depowers the room you're in, and you lack a blunt object to smash at least a window with assuming you have a window to escape through, (looking at various parts of meta and delta for this, meta's xenobio and viro intersection and delta's viro intersection, and that one room in pubby science), that's just how that goes. We can't expect everyone to be capable of getting out of that situation, so why should a zombie, a braindead predator, be able to accomplish what a greyshirt cannot?
> 
> It might not be fun for the zombies, but it's just as not fun for anyone. The inclusion of hands was at most hopefully going to make up for the loss of what is essentially a free pass to anywhere short of a location being blocked by a wall.
> 
> Maybe it's time to look at adding safety doors to more areas of the station. Like the public ones. Also, opening firelocks, though I believe the weedman himself vetoed making firelocks less unreasonable to escape through.

Imagine not rushing tools roundstart just for this reason B^)

**Comment by Jimmius** [comment]
> I acknowledge this is a netloss in just sheer braindead lethality, and for the most part that's what people care about, but if I can add a few more interactions that will make them more zombie-like, and fix the problems with them potentially being stranded, hopefully it will be a relatively neutral change.

I think that's why a lot of people (including myself) were concerned with the changes. They're expensive and pretty rare to see, so the idea of them basically just becoming green assistants who have to hack open doors instead of smashing them down wasn't exactly thrilling.
That said, they're definitely janky as hell right now, so making them more consistent without taking away what makes them actually fun to see (when the stars align and enough people get infected to be a serious threat) is good.

**Comment by wesoda25** [comment]
Zombies shouldn’t be able to use things like tools at all. Seems very odd.

**Comment by nemvar** [comment]
Can't wait for bola zombos.

**Comment by necromanceranne** [comment]
Well hopefully now, zombies can't hack airlocks, and additionally they now smash themselves against various objects to damage them significantly! This includes airlocks and windows!

They also now have strong grabber to really give it that zombie tackle feel. Getting bitten by a zombie knocks you over as well!

File: code/game/mecha/mecha.dm (hunk 1 above)
Comment (necromanceranne) [reply code/game/mecha/mecha.dm:892]: I can't allow this information to get out. I'm sorry, I must now close this.

**Comment by AdamElTablawy** [comment]
This kind of stops outbreaks in their tracks, given you can now bolt zombies in.

**Comment by necromanceranne** [comment]
There isn't a way to bolt them in anymore now that they can smash down airlocks. While airlocks are abundant, blast doors are not.

**Comment by Winterous** [comment]
Do they still use the default punching noise? Their attack verb should be changed too.

**Comment by necromanceranne** [comment]
No, they use the slash attack verb.

**Comment by brotherbeyondd** [comment]
Didn't you like, massively nerf aggressive grabs in an ided? Zombies would literally have to crit people in order to infect them.

**Comment by necromanceranne** [comment]
Strong grabber is pretty strong. It is quite literally old scarp grab and cqc grab. People keep talking about how I nerfed aggrograbs but they're still practically a hardstun that ends only based on rng, which is fairly strong. At minimum now, it's 40% to last 4 seconds, otherwise it is a disarm, and always does a big chunk of stamina damage.

Zombies can spam that. If you could pull off a stomach knee with old scarp, you can bite someone as a zombie. And likely you'll stamina crit someone.

**Comment by brotherbeyondd** [comment]
You do have to realise that zombies are literally any player that gets infected. Most people are generally a bit too stupid to do anything like that. The old zombie was simple for a simple antag, bite = infect, hit everything. This is making it needlessly complex for a very vague benefit? And with the grabs, two disarms are guaranteed to hardstun for four seconds whilst aggressive grabs are RNG based and are almost impossible to pull of in the middle of a fight without a martial art or a stun. I feel like this change is just not needed.

**Comment by Krysonism** [comment]
My worry is that zombie outbreaks will fizzle out before they even can start properly.

Strong grabber is pretty strong trait in the hands of a competent player though.

**Comment by necromanceranne** [comment]
It's not my fault if players don't read the flavour text telling you to grab people and then bite them. I don't think it a valid argument that adding one more step of operation to do your zombie thing is somehow going to ruin everything when, if the player in question was so fresh they can't change intents to bite someone, they probably can't kill someone while unbelievably slow either. (Maybe, now that I think about it, it would be a good idea to add a guaranteed bite if the person they're hitting is at least in crit, so they don't accidentally forget if they do manage to swarm someone to death)

Also I just said their grabs are like the martial art grab. It's the trait leather golems have, in fact. There isn't anything that made martial art grabs special compared to normal aggressive grabs. They work the same way, with the same rules, only that they elevate immediately to aggrograbs.

File: code/game/objects/structures/window.dm [hunk 3]
```diff
@@ -143,10 +143,26 @@
...
-		"<span class='notice'>You knock on [src].</span>")
+	if(ishuman(user) && density)
+		var/mob/living/carbon/human/H = user
+		if((HAS_TRAIT(H, TRAIT_DUMB)) && Adjacent(user))
+			playsound(src, 'sound/effects/bang.ogg', 25, TRUE)
+			user.do_attack_animation(src, ATTACK_EFFECT_PUNCH)
+			if(!istype(H.head, /obj/item/clothing/head/helmet))
+				H.visible_message("<span class='danger'>[user] headbutts the [src].</span>", \
+									"<span class='userdanger'>You headbutt the [src]!</span>")
+				H.Paralyze(100)
```
Comment (Fikou) [inline code/game/objects/structures/window.dm:154]: is this a joke 🥴

Comment (necromanceranne) [reply code/game/objects/structures/window.dm:154]: It's actually the same check airlocks already have. >w>

**Comment by necromanceranne** [comment]
Zombies are now terrible at surgery, to make sure they don't get up to some truly terrible mischief, like self curing.

Additionally, zombies also now bite you while you're unable to protect yourself! So if you're cuffed to a chair, zombies can immediately bite! Or if you're stunned or in critical for that matter. Don't be afraid to throw hands, basically.

**Comment by ExcessiveUseOfCobblestone** [comment]
this needs to be done like a whitelist of what zombies CAN do rather than a blacklist of what zombies cant, otherwise I'd rather just have normal claws with like different attack verbs and sounds if the issue is they dont "bite"

**Comment by Winterous** [comment]
> 
> 
> this needs to be done like a whitelist of what zombies CAN do rather than a blacklist of what zombies cant

I think it makes sense for practically any interaction to have a check of "are you physically or mentally capable of doing it" like with golems being unable to fire guns, or monkeys being unable to operate machinery.

Like, are both your arms crippled? You can't do much with your hands then. Are you wearing boxing gloves? Again, can't do much with your hands. But if you're wearing METAL boxing gloves you can do even less, can't even hold items.

Would make it a lot easier to restrict what something can and can't do based on its inherent abilities. X creature has low intelligence, nimble hands. Y creature has high intelligence, stumps for hands. Zombies are both low intelligence and poor manipulation; they can push buttons and manipulate objects, but can't do anything more advanced than simple tool use.

**Comment by optimumtact** [comment]
zombies have forced weapons to prevent them doing unzombie things.

I don't mind the bite mechanics, but the rest of this basically has to go

**Comment by MrDoomBringer** [comment]
based orange

**Comment by Winterous** [comment]
Should a grappling mechanic be added to the claws, then? Maybe they could have a pseudo-tackle, lunge forward quickly and try to grab on.

**Comment by optimumtact** [comment]
tbh I don't even know, who cares, they're just some meme zombies, delete them from the game I say
ignore my previous comment I don't evne care

**Pull request closed without merging**